src/services/profiles/medications.ts

```diff
@@ -0,0 +1,68 @@
+import type { HttpInstance } from '../../types';
+import {
+  AffectedRecords,
+  ObjectId,
+  PagedResult,
+  Results,
+  ResultResponse,
+} from '../types';
+import { Medication } from './types';
+
+export default (client, httpAuth: HttpInstance) => ({
+  /**
+   * Retrieve a list of all the defined medications
+   * Permission | Scope | Effect
+   * - | - | -
+   * none | | Everyone can use this endpoint
+   *
+   * @returns PagedResult<Medication>
+   */
+  async getMedications(): Promise<PagedResult<Medication>> {
+    return (await client.get(httpAuth, '/medication')).data;
+  },
+
+  /**
+   * Add a new medicine to a specified profile
+   * Permission | Scope | Effect
+   * - | - | -
+   * `UPDATE_PROFILES` | `staff enlistment` | Add a new medicine to a profile for this group
+   * `UPDATE_PROFILES` | `global` | Add a new medicine to a profile for any group
+   *
+   * @param profileId Id of the targeted profile
+   * @param requestBody Medication data
+   * @returns Medication
+   * @throws {ResourceUnknownError}
+   */
+  async create(
+    profileId: ObjectId,
+    requestBody: Medication
```

Comment:
`medication_frequency` is a required field, but it's not correctly shown in swagger

Reply:
I think number and unit are also required, but not in swagger and also not in the response. But if you do not pass them in -> 500 error